packages/dds/tree/src/core/tree/delta.ts

```diff
@@ -172,57 +169,51 @@ export type Skip = number;
  * Describes modifications made to a subtree.
  * @alpha
  */
-export interface Modify<TTree = ProtoNode> {
-	readonly type: typeof MarkType.Modify;
-	readonly setValue?: Value;
+export interface HasModifications<TTree = ProtoNode> {
 	readonly fields?: FieldMarks<TTree>;
+	/**
+	 * When set, indicates the new value that should be assigned to the node.
+	 * Can be set to `undefined` to convey that the node's value should be cleared.
+	 * Readers of this field should use the following check to distinguish the above cases:
+	 * `Object.prototype.hasOwnProperty.call(mark, "setValue")`
+	 */
+	readonly setValue?: Value;
 }
 
 /**
- * Describes the deletion of a contiguous range of node.
+ * Describes modifications made to an otherwise untouched subtree.
  * @alpha
  */
-export interface Delete {
-	readonly type: typeof MarkType.Delete;
-	readonly count: number;
+export interface Modify<TTree = ProtoNode> extends HasModifications<TTree> {
+	readonly type: typeof MarkType.Modify;
 }
 
 /**
- * Describes the deletion of a single node.
- * Includes descriptions of the modifications the node.
+ * Describes the deletion of a contiguous range of node.
  * @alpha
  */
-export interface ModifyAndDelete<TTree = ProtoNode> {
-	readonly type: typeof MarkType.ModifyAndDelete;
-	readonly fields: FieldMarks<TTree>;
+export interface Delete<TTree = ProtoNode> extends HasModifications<TTree> {
+	readonly type: typeof MarkType.Delete;
+	/**
+	 * Must be 1 when either `setValue` or `fields` is populated.
+	 */
+	readonly count: number;
 }
 
 /**
  * Describes the moving out of a contiguous range of node.
  * @alpha
  */
-export interface MoveOut {
+export interface MoveOut<TTree = ProtoNode> extends HasModifications<TTree> {
 	readonly type: typeof MarkType.MoveOut;
-	readonly count: number;
 	/**
-	 * The delta should carry exactly one `MoveIn` mark with the same move ID.
+	 * Must be 1 when either `setValue` or `fields` is populated.
```

Comment:
I don't think I'm grasping what you're proposing.
The reason the `count` has to be 1 when either `setValue` or `fields` is populated is that when that's the case, the mark is describing the modifications applied to a single node. If the `count` were greater than 1, say, 2 for example, we would not know which of the two nodes the `setValue` or `fields` fields apply to.

Reply:
I was thinking that if a skip is included, it could be used to select which node to apply the modify to. Like if it were 1 in your example, we would know to apply it to the second node. Although, that probably conflicts with some of the other marks. It just feels kind of eh to me to have this requirement without the API enforcing it. We don't ever expect SharedTree consumers to deal with this API though, right?

Reply:
I see. I agree we could take that approach. It's not clear to me whether it's worth the increase in format and code complexity. It also would only help so much because you could have 10 moved-out nodes, 3 of which are also being modified, in which case you still have to split up the mark in some way. We could represent an array of skips and modifications for a given MoveOut/Delete/Insert/Skip but that's an even bigger increase in complexity.

As you pointed out, this format is not really meant for public consumption so I don't feel too bad about the lack of type-enforcement here.